packages/runtime-common/schema-analysis-plugin.ts

```diff
@@ -41,179 +38,133 @@ export interface PossibleField {
+            }
           }
         },
       },
       ClassDeclaration: {
         enter(path: NodePath<t.ClassDeclaration>, state: State) {
-          if (!path.node.superClass) {
-            return;
-          }
-
-          let sc = path.get('superClass');
-          if (sc.isReferencedIdentifier()) {
-            let classRef = makeClassReference(path.scope, sc.node.name, state);
-            if (classRef) {
-              state.insideCard = true;
-              let exportedAs: string | undefined;
-              let { parentPath } = path;
-              let localName = path.node.id ? path.node.id.name : undefined;
-              if (parentPath.isExportNamedDeclaration()) {
-                // the class declaration is part of a named export
-                exportedAs = localName;
-              } else if (parentPath.isExportDefaultDeclaration()) {
-                // the class declaration is part of a default export
-                exportedAs = 'default';
-              } else {
-                // the class's identifier is referenced in a node whose parent is an ExportSpecifier
-                let binding = localName
-                  ? path.scope.getBinding(localName)
-                  : undefined;
-                if (binding) {
-                  let maybeExportSpecifierLocal = binding.referencePaths.find(
-                    (b) => b.parentPath?.isExportSpecifier(),
-                  ) as NodePath<t.Identifier> | undefined;
-                  if (maybeExportSpecifierLocal) {
-                    exportedAs = getName(
-                      (
-                        maybeExportSpecifierLocal.parentPath as NodePath<t.ExportSpecifier>
-                      ).node.exported,
-                    );
+          let maybeCard = lookForCard(path, state);
+          if (maybeCard) {
+            state.opts.possibleCards.push(maybeCard);
+            maybeCard.path.traverse({
+              Decorator: {
+                enter(path: NodePath<t.Decorator>) {
+                  let expression = path.get('expression');
+                  if (!expression.isIdentifier()) {
+                    return;
+                  }
+                  let decoratorInfo = getNamedImportInfo(
+                    path.scope,
+                    expression.node.name,
+                  );
+                  if (!decoratorInfo) {
+                    return; // our @field decorator must originate from a named import
                   }
-                }
-              }
-
-              state.opts.possibleCards.push({
-                super: classRef,
-                localName,
-                path,
-                possibleFields: new Map(),
-                exportedAs,
-              });
-            }
-          }
-        },
-
-        exit(_path: NodePath<t.ClassDeclaration>, state: State) {
-          state.insideCard = false;
-        },
-      },
 
-      Decorator(path: NodePath<t.Decorator>, state: State) {
```

Comment:
ive moved this into a nested traversal when we've found a card by iterating thru the class. Only applicable to possibleCards

Reply:
Reason I did this is so that we don't use global state insideCard that can affect multiple traversals